nixops/util.py

```diff
@@ -398,26 +398,50 @@ def ping_tcp_port(
     return False
 
 
-def wait_for_tcp_port(
-    ip: str,
-    port: int,
-    timeout: int = -1,
-    open: bool = True,
-    callback: Optional[Callable[[], Any]] = None,
+def wait_for_success(
+    fn: Callable, timeout: int = -1, callback: Optional[Callable[[], Any]] = None,
```

Comment:
```suggestion
    fn: Callable, timeout: Optional[int] = None, callback: Optional[Callable[[], Any]] = None,
```

Let's use our type system to our advantage here, and in the next function too.